File: NRZ.Web/client/src/models/seizedAvailabiityReport/vessel.ts
export default class Vessel {
    bt: string;
    nt: string;
    maxLength: string;
    lengthBetweenPerpendiculars: string;
    maxWidth: string;
    waterplane: string;
    shipboardHeight: string;
    deadWeight: string;
    numberOfEngines: string;
    enginesFuel: string;
    sumEnginePower: string;
    bodyNumber: string;

    constructor(obj: any = {}) {
        this.bt = obj.bt || '';
        this.nt = obj.nt || '';
        this.maxLength = obj.maxLength || '';
        this.lengthBetweenPerpendiculars = obj.lengthBetweenPerpendiculars || '';
        this.maxWidth = obj.maxWidth;
        this.waterplane = obj.waterplane || '';
        this.shipboardHeight = obj.shipboardHeight || '';
        this.deadWeight = obj.deadWeight || '';
        this.numberOfEngines = obj.numberOfEngines || '';
        this.enginesFuel = obj.enginesFuel || '';
        this.sumEnginePower = obj.sumEnginePower || '';
        this.bodyNumber = obj.bodyNumber || '';
    }
}

File: NRZ.Data/Vessel.cs
using System;
using System.Collections.Generic;

namespace NRZ.Data
{
    public partial class Vessel
    {
        public Vessel()
        {
            Distraint = new HashSet<Distraint>();
            VesselEngine = new HashSet<VesselEngine>();
            VesselOwner = new HashSet<VesselOwner>();
            VesselRegistrationData = new HashSet<VesselRegistrationData>();
        }

        public int Id { get; set; }
        public decimal? Bt { get; set; }
        public decimal? Nt { get; set; }
        public decimal? MaxLength { get; set; }
        public decimal? LengthBetweenPerpendiculars { get; set; }
        public decimal? MaxWidth { get; set; }
        public decimal? Waterplane { get; set; }
        public decimal? ShipboardHeight { get; set; }
        public decimal? Deadweight { get; set; }
        public int? NumberOfEngines { get; set; }
        public string EnginesFuel { get; set; }
        public decimal? SumEnginePower { get; set; }
        public string BodyNumber { get; set; }

        public virtual VesselExtension VesselExtension { get; set; }
        public virtual ICollection<Distraint> Distraint { get; set; }
        public virtual ICollection<VesselEngine> VesselEngine { get; set; }
        public virtual ICollection<VesselOwner> VesselOwner { get; set; }
        public virtual ICollection<VesselRegistrationData> VesselRegistrationData { get; set; }
    }
}
